dbt/include/snowflake/macros/materializations/incremental.sql

```diff
@@ -10,10 +10,18 @@
        for faster overall incremental processing.
   #} */
```

Comment:
@McKnight-42 we should update the comments in L3-L11 to align with our changes.

Maybe something generally like the following? There's lots of different ways we could write the comments here. A key thing to communicate is _why_ different strategies use either a view or a temporary table. It would be nice if we make it explicit rather than having to do detective work, especially in the case where `unique_key` is none.

(I didn't bother to limit the line lengths below while we are iterating on the exact wording, but we'll want to add appropriate new lines in the final commit.)

```
  /* {#
       High-level principles:
       If we are running multiple statements (DELETE + INSERT),
       and we want to guarantee identical inputs to both statements, then
       we _must_ first save the model query results as a temporary table (which presumably comes with a performance cost).
       If we are running a single statement (MERGE or INSERT alone),
       we _may_ save the model query definition as a view instead,
       for (presumably) faster overall incremental processing.

       Low-level specifics:
       Languages other than SQL (like Python) will use a temporary table.
       With the `default` strategy of `merge`, the user may choose between a temporary table and view (defaulting to view).
       The `append` strategy can use a view because it will run a single INSERT statement.
       When the `unique_key` is none, then we can use a view because it will run a single INSERT statement.
       Otherwise, play it safe by using a temporary table.
  #} */
```

Reply:
Sounds great! will push up the change to the wording asap

Reply:
Awesome @McKnight-42 !

Could you choose a maximum line length for the comments and then apply it? It looks like it was using a max line length of 70 previously. Using either [80](https://stackoverflow.com/questions/578059/studies-on-optimal-code-width) or [88](https://black.readthedocs.io/en/stable/the_black_code_style/current_style.html#line-length) seems reasonable also.